diff --git a/hack/build/init-buildx.sh b/hack/build/init-buildx.sh
index b896f5a3..0459224c 100755
--- a/hack/build/init-buildx.sh
+++ b/hack/build/init-buildx.sh
@@ -22,7 +22,8 @@ current_builder="$(docker buildx inspect)"
 # linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
 if ! grep -q "^Driver: docker$" <<<"${current_builder}" && \
      grep -q "linux/amd64" <<<"${current_builder}" && \
-     grep -q "linux/arm64" <<<"${current_builder}"; then
+     grep -q "linux/arm64" <<<"${current_builder}" && \
+     grep -q "linux/ppc64le" <<<"${current_builder}"; then
   exit 0
 fi
 
diff --git a/images/base/Dockerfile b/images/base/Dockerfile
index 1ce80d74..fac12de0 100644
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -108,10 +108,10 @@ COPY --chmod=0755 scripts/third_party/gimme/gimme /usr/local/bin/
 COPY --chmod=0755 scripts/target-cc /usr/local/bin/
 # tools needed at build-time only
 # first ensure we can install packages for both architectures
-RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 \
-    && clean-install bash ca-certificates curl git make pkg-config \
-    crossbuild-essential-amd64 crossbuild-essential-arm64 \
-    libseccomp-dev:amd64 libseccomp-dev:arm64
+RUN dpkg --add-architecture ppc64el \
+    && clean-install bash ca-certificates curl git make pkg-config gcc \
+    crossbuild-essential-amd64 crossbuild-essential-ppc64el \
+    libseccomp-dev:amd64 libseccomp-dev:ppc64el
 # set by makefile to .go-version
 ARG GO_VERSION
 RUN eval "$(gimme "${GO_VERSION}")" \
diff --git a/images/base/scripts/target-cc b/images/base/scripts/target-cc
index f8226e15..dc9980f1 100755
--- a/images/base/scripts/target-cc
+++ b/images/base/scripts/target-cc
@@ -23,6 +23,8 @@ case $TARGETARCH in
     echo -n 'aarch64-linux-gnu-gcc' ;;
   amd64)
     echo -n 'x86_64-linux-gnu-gcc' ;;
+  ppc64le)
+    echo -n 'powerpc64le-linux-gnu-gcc' ;;
   *)
     exit 1 ;;
 esac
diff --git a/images/local-path-provisioner/Dockerfile b/images/local-path-provisioner/Dockerfile
index df601536..e2050319 100644
--- a/images/local-path-provisioner/Dockerfile
+++ b/images/local-path-provisioner/Dockerfile
@@ -15,7 +15,7 @@
 # NOTE the actual go version will be overridden
 FROM --platform=$BUILDPLATFORM docker.io/library/golang:latest
 COPY --chmod=0755 scripts/third_party/gimme/gimme /usr/local/bin/
-RUN git clone --filter=tree:0 https://github.com/rancher/local-path-provisioner
+RUN git clone --filter=tree:0 --branch=ppc64le-support https://github.com/kishen-v/local-path-provisioner
 ARG VERSION
 # set by makefile to .go-version
 # TODO: scripts/build builds for multiple platforms, so we waste a little time here
@@ -23,7 +23,7 @@ ARG TARGETARCH GO_VERSION
 RUN eval "$(gimme "${GO_VERSION}")" \
     && export GOTOOLCHAIN="go${GO_VERSION}" \
     && cd local-path-provisioner \
-    && git fetch && git checkout "${VERSION}" \
+#    && git fetch && git checkout "${VERSION}" \
     && GOARCH=$TARGETARCH scripts/build \
     && mv bin/local-path-provisioner-$TARGETARCH /usr/local/bin/local-path-provisioner \
     && GOBIN=/usr/local/bin go install github.com/google/go-licenses@latest \
diff --git a/pkg/build/nodeimage/const_cni.go b/pkg/build/nodeimage/const_cni.go
index a1c870c9..4568afbd 100644
--- a/pkg/build/nodeimage/const_cni.go
+++ b/pkg/build/nodeimage/const_cni.go
@@ -20,7 +20,7 @@ package nodeimage
 The default CNI manifest and images are our own tiny kindnet
 */
 
-const kindnetdImage = "docker.io/kindest/kindnetd:v20250214-acbabc1a"
+const kindnetdImage = "quay.io/powercloud/kind-kindnetd:v20240626-0296c52"
 
 var defaultCNIImages = []string{kindnetdImage}
 
diff --git a/pkg/build/nodeimage/const_storage.go b/pkg/build/nodeimage/const_storage.go
index 0f4806bf..c887b3fe 100644
--- a/pkg/build/nodeimage/const_storage.go
+++ b/pkg/build/nodeimage/const_storage.go
@@ -26,8 +26,8 @@ NOTE: we have customized it in the following ways:
 - tolerate control plane scheduling taints
 */
 
-const storageProvisionerImage = "docker.io/kindest/local-path-provisioner:v20250214-acbabc1a"
-const storageHelperImage = "docker.io/kindest/local-path-helper:v20241212-8ac705d0"
+const storageProvisionerImage = "quay.io/powercloud/kind-local-path-provisioner:v20240626-0296c52"
+const storageHelperImage = "quay.io/powercloud/kind-local-path-helper:v20240626-0296c52"
 
 // image we need to preload
 var defaultStorageImages = []string{storageProvisionerImage, storageHelperImage}
diff --git a/pkg/cluster/internal/loadbalancer/const.go b/pkg/cluster/internal/loadbalancer/const.go
index 3600b338..e84bf8c4 100644
--- a/pkg/cluster/internal/loadbalancer/const.go
+++ b/pkg/cluster/internal/loadbalancer/const.go
@@ -17,7 +17,7 @@ limitations under the License.
 package loadbalancer
 
 // Image defines the loadbalancer image:tag
-const Image = "docker.io/kindest/haproxy:v20230606-42a2262b"
+const Image = "quay.io/powercloud/kind-haproxy:v20240626-0296c52"
 
 // ConfigPath defines the path to the config file in the image
 const ConfigPath = "/usr/local/etc/haproxy/haproxy.cfg"
